Test failure for `TestPoisson.test_mindist` in Azure CI job

See [this CI log](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=19489&view=logs&j=203f29d8-a99b-512b-bad4-9c985ff1792a&t=0e9bc6fa-f840-5c44-330b-bc382f4fd289)

This test is already marked as slow and as xfail on 32-bit. Now it fails here too - a 1.5GB array should not be necessary to test here, it'd be good to reduce the size by a factor of 4; that speeds up the test and avoids spurious failures on low-memory machines.

```
__________________________ TestPoisson.test_mindist ___________________________
[gw1] win32 -- Python 3.8.10 C:\hostedtoolcache\windows\Python\3.8.10\x64\python.exe
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\scipy\stats\tests\test_qmc.py:832: in test_mindist
    engine = self.qmce(
        d          = 4
        dimensions = [1, 3, 4]
        engine     = <scipy.stats._qmc.PoissonDisk object at 0x000001F05C8921F0>
        gen        = <itertools.product object at 0x000001F05CD8D640>
        high       = 0.1
        hypersphere = 'surface'
        hypersphere_methods = ['volume', 'surface']
        low        = 0.01
        ns         = 50
        radii      = array([0.04234447, 0.02299661, 0.07843557, 0.02020188, 0.09678935])
        radius     = 0.020201882000181155
        rng        = Generator(PCG64) at 0x1E84E1EE200
        sample     = array([[0.18997821, 0.64813041, 0.22948406, 0.36861593],
       [0.18082003, 0.61758709, 0.22225516, 0.38814378],
    ...45],
       [0.21233569, 0.65256325, 0.21749   , 0.33823402],
       [0.22073226, 0.62634524, 0.24930688, 0.29894718]])
        self       = <scipy.stats.tests.test_qmc.TestPoisson object at 0x000001E82C8A6CA0>
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\scipy\stats\_qmc.py:1759: in __init__
    self._initialize_grid_pool()
        __class__  = <class 'scipy.stats._qmc.PoissonDisk'>
        d          = 4
        hypersphere = 'surface'
        hypersphere_sample = {'surface': <bound method PoissonDisk._hypersphere_surface_sample of <scipy.stats._qmc.PoissonDisk object at 0x000001F...: <bound method PoissonDisk._hypersphere_volume_sample of <scipy.stats._qmc.PoissonDisk object at 0x000001F05C892B50>>}
        ncandidates = 30
        optimization = None
        radius     = 0.020201882000181155
        seed       = Generator(PCG64) at 0x1E84E1EE200
        self       = <scipy.stats._qmc.PoissonDisk object at 0x000001F05C892B50>
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\scipy\stats\_qmc.py:1766: in _initialize_grid_pool
    self.sample_grid = np.empty(
E   numpy.core._exceptions._ArrayMemoryError: Unable to allocate 1.49 GiB for an array with shape (100, 100, 100, 100, 4) and data type float32
        self       = <scipy.stats._qmc.PoissonDisk object at 0x000001F05C892B50>
```

--- a/scipy/stats/tests/test_qmc.py
+++ b/scipy/stats/tests/test_qmc.py
@@ -814,13 +814,11 @@ class TestPoisson(QMCEngineTests):
         assert len(sample) <= ns * 2
         assert l2_norm(sample) >= radius
 
-    @pytest.mark.slow
-    @pytest.mark.xfail_on_32bit("Can't create large array for test")
     def test_mindist(self):
         rng = np.random.default_rng(132074951149370773672162394161442690287)
         ns = 50
 
-        low, high = 0.01, 0.1
+        low, high = 0.08, 0.2
         radii = (high - low) * rng.random(5) + low
 
         dimensions = [1, 3, 4]

